src/db.rs

```diff
@@ -22,6 +23,16 @@ impl WriteBatch {
         self.spending_rows.sort_unstable();
         self.txid_rows.sort_unstable();
     }
+    pub(crate) fn merge(mut self, other: WriteBatch) -> Self {
+        self.header_rows.extend(other.header_rows.into_iter());
+        self.funding_rows.extend(other.funding_rows.into_iter());
+        self.spending_rows.extend(other.spending_rows.into_iter());
+        self.txid_rows.extend(other.txid_rows.into_iter());
+        if self.height < other.height {
+            self.tip_row = other.tip_row
```

Comment:
Should we also update `self.height`?

Reply:
I don't think so ->, https://github.com/romanz/electrs/pull/931#discussion_r1310484987

Which is not ideal at all, but I am not sure if there are other options, or if I should keep this and add a comment